File: packages/chat-python/src/rfnry_chat_protocol/tenant.py
from __future__ import annotations

TenantScope = dict[str, str]


def matches(thread_tenant: TenantScope, identity_tenant: TenantScope) -> bool:
    for key, value in thread_tenant.items():
        iv = identity_tenant.get(key)
        if iv == "*":
            continue
        if iv != value:
            return False
    return True

File: packages/chat-typescript/src/tenant.ts
export type TenantScope = Record<string, string>

export function matches(threadTenant: TenantScope, identityTenant: TenantScope): boolean {
  for (const [key, value] of Object.entries(threadTenant)) {
    const iv = identityTenant[key]
    if (iv === '*') continue
    if (iv !== value) return false
  }
  return true
}
